cpp/src/io/comp/comp.hpp

```diff
@@ -57,5 +57,57 @@ std::vector<uint8_t> compress(compression_type compression,
                               host_span<uint8_t const> src,
                               rmm::cuda_stream_view stream);
 
+/**
+ * @brief Maximum size of uncompressed blocks that can be compressed.
+ *
+ * @param compression Compression type
+ * @returns maximum block size
+ */
+[[nodiscard]] std::optional<size_t> compress_max_allowed_block_size(compression_type compression);
+
+/**
+ * @brief Gets input and output alignment requirements for the given compression type.
+ *
+ * @param compression Compression type
+ * @returns required alignment
+ */
+[[nodiscard]] size_t compress_required_block_alignment(compression_type compression);
```

Comment:
Not that I have a better name, but just sharing my first impression: "blocks" seems to be an ORC-specific term, and using it in Parquet code feels a bit odd.

Reply:
for some reason I thought that nvCOMP uses the term blocks. I'll switch these to chunks, to at least align with nvCOMP.